js/task17.js

```diff
@@ -0,0 +1,124 @@
+const ul = document.getElementById('lists');
+
+let currentImage = 0;
+const imagesArr = [];
+
+function createLoader() {
+	const loaderContainer = document.createElement('li');
+	const loader = document.createElement('img');
+	loader.src = './img/loading-circle.gif';
+	ul.appendChild(loader).appendChild(loaderContainer);
+
+	loader.classList.add('loader_image');
+	return loader;
+}
+
+function fetchData() {
+	return new Promise((resolve, reject) => {
+		setTimeout(() => {
+			resolve(fetch('./js/json/task17.json'));
+		}, 3000);
+	});
+}
+
+async function getData() {
+	const loaderImage = createLoader();
+
+	const li = document.createElement('li');
+	li.classList.add('loader');
+
+	ul.appendChild(li).appendChild(loaderImage);
+
+	let response, res, data;
+
+	try {
+		response = await fetchData();
+		res = await response.json();
+		data = res.data;
+
+		return data;
+	} catch (error) {
+		ul.innerHTML = "Couldn't get response.";
+	} finally {
+		console.log('You got the result.');
+
+		data.forEach((el, index) => {
+			const fragment = document.createDocumentFragment();
+			const image = document.createElement('img');
+			const li = document.createElement('li');
+
+			image.src = el.image;
+			li.appendChild(image);
+			imagesArr.push(li);
+
+			li.classList.add('slide-images');
+
+			if (index === 0) {
+				li.classList.add('z-index-100');
+			}
+			fragment.appendChild(li);
+			ul.appendChild(fragment);
+		});
+	}
+}
+getData();
+
+function displayPrevImage(currentImage) {
+	const imageShown = imagesArr[currentImage];
+	const prevImage = imagesArr[currentImage + 1];
+
+	imageShown.classList.add('z-index-100');
+	prevImage.classList.remove('z-index-100');
+}

```

Comment:
IMO :

I thought "prevImage" was confusing.
`const lastImage `might be more appropriate.

Also, currentImage is a global variable, so it may not need to be an argument.

Or you could do the currentImage decrement here, as follows.
```suggestion
function displayPrevImage() {
  imagesArr[currentImage].classList.remove('z-index-100');
  currentImage--;
  imagesArr[currentImage].classList.add('z-index-100');	
}

```

Reply:
>I thought "prevImage" was confusing.
const lastImage might be more appropriate.

Great suggestion. I'll take it.

Reply:
I updated those two functions. [16ba3c4 ](https://github.com/suefrontend/kenmori-handson/pull/18/commits/16ba3c4b3423e57845e696bf0973fe84cee05d4c)